src/main/java/Duke.java

```diff
@@ -1,10 +1,136 @@
+                        StringBuilder sb = new StringBuilder();
+                        for (int i = 0; i < tasks.getTaskListSize(); i++) {
+                            System.out.println(i + 1 + ". " + tasks.getTask(i).toString());
+                            sb.append(tasks.getTask(i).saveToList() + "\n");
+                        }
+                        saveToFile.usingFileWriter(sb.toString());
+                        ui.showLine();
+                    }
+                } else if (arrString[0].equalsIgnoreCase("done")) {
+                    try {
+                        int taskNumber = Integer.parseInt(arrString[1].strip()) - 1;
+                        if (taskNumber >= 0 && taskNumber < tasks.getTaskListSize()) {
+                            tasks.getTask(taskNumber).doneTask();
+                            ui.doneTask();
+                            System.out.println(tasks.getTask(taskNumber).toString());
+                        } else {
+                            ui.invalidTask();
+                        }
+                        ui.showLine();
+                    } catch (ArrayIndexOutOfBoundsException ex) {
+                        throw new DukeException(ui.missingTaskNumber());
+                    }
+
+                } else if (arrString[0].equalsIgnoreCase("delete")) {
+                    try {
+                        int taskNumber = Integer.parseInt(arrString[1].strip()) - 1;
+                        if (taskNumber >= 0 && taskNumber < tasks.getTaskListSize()) {
+                            ui.deletedTask();
+                            System.out.println(tasks.getTask(taskNumber).toString());
+                            tasks.deleteTask(taskNumber);
+                            ui.taskInList(tasks.getTaskListSize());
+                        } else {
+                            ui.invalidTask();
+                        }
+                        ui.showLine();
+                    } catch (ArrayIndexOutOfBoundsException ex) {
+                        throw new DukeException(ui.missingTaskNumber());
+                    }
+
+                } else if (arrString[0].equalsIgnoreCase("todo")) {
+                    try {
+                        Todo todo = new Todo(arrString[1]);
+                        tasks.addTask(todo);
+                        ui.addedCommand();
+                        System.out.println(todo.toString());
+                        int listSize = tasks.getTaskListSize();
+                        ui.taskInList(listSize);
+                    } catch (ArrayIndexOutOfBoundsException ex) {
+                        throw new DukeException(ui.incompleteCommand("Todo"));
+                    }
+                } else if (arrString[0].equalsIgnoreCase("event")) {
+                    try {
+                        String[] eventString = arrString[1].split("/");
+                        Event event = new Event(eventString[0].strip(), eventString[1].substring(2).strip());
```

Comment:
I like that you stripped off trailing whitespaces to keep only the essential user input.